src/java/com/puppetlabs/puppetserver/ShellUtils.java

```diff
@@ -0,0 +1,88 @@
+package com.puppetlabs.puppetserver;
+
+import org.apache.commons.exec.DefaultExecutor;
+import org.apache.commons.exec.PumpStreamHandler;
+import org.apache.commons.exec.CommandLine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+public class ShellUtils {
+
+    private static final Logger log = LoggerFactory.getLogger(ShellUtils.class);
+
+    /**
+     * Takes a prepared CommandLine instance and executes it using some sane
+     * defaults and a DefaultExecutor. Also makes a delicious pancake.
+     *
+     * @param commandLine CommandLine instance to execute
+     * @return An ExecutionResult with output[String], error[String], and
+     *                  the exit code[Integer] of the process
+     *
+     * @throws InterruptedException
+     * @throws IOException
+     */
+    private static ExecutionResult executeExecutor(CommandLine commandLine)
+            throws InterruptedException, IOException {
+        DefaultExecutor executor = new DefaultExecutor();
+        ByteArrayOutputStream errStream = new ByteArrayOutputStream();
+        ByteArrayOutputStream outStream = new ByteArrayOutputStream();
+        PumpStreamHandler streamHandler = new PumpStreamHandler(outStream, errStream);
+
+        // Don't throw exception on non-zero exit code
+        executor.setExitValues(null);
+
+        // Set up the handlers
+        executor.setStreamHandler(streamHandler);
+
+        Integer exitCode = executor.execute(commandLine);
```

Comment:
Does the executor have a timeout? Is it configurable? Might we want end-users to be able to set a timeout?